services/user/CommonSys/common/packages/wasm-loader/src/main.ts

```diff
@@ -24,68 +21,53 @@ interface Importables {
     [key: string]: string;
 }
 
-const runWasm = async (
-    wasm: ArrayBuffer,
-    importables: Importables[],
-    method: string,
-    params: any[]
-) => {
-    console.info("runWasm().top 2");
-    const { load } = await import("rollup-plugin-wit-component");
-
-    console.time("Load");
-    const mod = await load(
-        // @ts-expect-error fff
-        wasm,
-        importables
-    );
-    console.timeEnd("Load");
-
-    return mod[method](...params);
-};
+const wasmUrlToIntArray = (url: string) =>
+    fetch(url)
+        .then((res) => res.arrayBuffer())
+        .then((buffer) => new Uint8Array(buffer));
 
 const functionCall = async ({
     id,
     args,
     precomputedResults
 }: PluginCallPayload) => {
-    const url = "/plugin.wasm";
+    const [pluginBytes, parserBytes, wasmComponent] = await Promise.all([
+        wasmUrlToIntArray("/plugin.wasm"),
+        wasmUrlToIntArray("/common/wasm-loader/component_parser.wasm"),
```

Comment:
Where does this file come from? When I build and run, I'm getting a 404 on this file.

Reply:
It's supposed to be a build artifact, but @Velua you somehow lost the project that generates the wasm when you migrated into this new branch. I [still see it](https://github.com/gofractally/psibase/pull/592/files#diff-8b1fea070d00786d2ef9c890d5b628b4e657ab912bff0675d8436188ad9037d8) in the branch in your other open PR

Reply:
@brandonfancher and @Velua.
I've added the component-parser in [a separate PR](https://github.com/gofractally/psibase/pull/638). Once merged, it will be available on chain by default.

@Velua it implies two changes you need to make this pr:
1. Undo the CMakeLists.txt changes
2. Anything that attempts to load the component_parser from chain should look for it at: `/common/component_parser.wasm`. It's not actually coupled to the loader, hence the separation.